app/models/user.rb

```diff
@@ -11,4 +11,26 @@ class User < ApplicationRecord
 
   enum active_status: {active: 0, inactive: 1}
   enum donor_status: {donor: 0, recipient: 1}
+  before_save :normalize_phone
+
+
+ def formatted_phone
+   parsed_phone = Phonelib.parse(phone_number)
+   return phone_number if parsed_phone.invalid?
+
+   formatted =
+     if parsed_phone.country_code == "1"
+       parsed_phone.full_national
+     else
+       parsed_phone.full_international
+     end
+   formatted.gsub!(";", " x")
+   formatted
+ end
+
+ private
+
+ def normalize_phone
+   self.phone_number = Phonelib.parse(phone_number).full_e164.presence
+ end
```

Comment:
This was super cool to see, thanks for walking me through how you built this out!